### PR TITLE
fix(frontend): improve the check of duplicated relations

### LINKED ISSUE #5783: fix:  improve the process of detecting duplicate relation names
Now there is a function `check_relation_name_duplicated` in frontend, which is used to detecting duplicate sources' and tables' names. It misses the process of detecting duplication of sinks' and tables' (not mvs') names. (When creating index, it will use the index table to detect this.)
And meta doesn't have the process. We should add it to meta to avoid conflicts between multiple frontends.

### PR DESCRIPTION
**Pull request by Gun9niR**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Check duplicated table at FE
- Check relation name against all kinds of relations at meta

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)

Closes #5783

### CODE REVIEW

File: src/frontend/src/handler/create_table.rs [hunk 1]
```diff
@@ -267,6 +268,24 @@ pub async fn handle_create_table(
     constraints: Vec<TableConstraint>,
 ) -> Result<RwPgResponse> {
     let session = context.session_ctx.clone();
+    {
+        let db_name = session.database();
+        let catalog_reader = session.env().catalog_reader().read_guard();
+        let (schema_name, table_name) = {
+            let (schema_name, source_name) =
```
Comment (HuaHuaY) [inline src/frontend/src/handler/create_table.rs:275]:
```suggestion
            let (schema_name, table_name) =
```

File: src/frontend/src/handler/create_table.rs [hunk 2]
```diff
@@ -267,6 +268,24 @@ pub async fn handle_create_table(
...
+                Binder::resolve_table_or_source_name(db_name, table_name.clone())?;
+            let search_path = session.config().get_search_path();
+            let user_name = &session.auth_context().user_name;
+            let schema_name = match schema_name {
+                Some(schema_name) => schema_name,
+                None => catalog_reader
+                    .first_valid_schema(db_name, &search_path, user_name)?
+                    .name(),
+            };
+            (schema_name, source_name)
```
Comment (HuaHuaY) [inline src/frontend/src/handler/create_table.rs:285]:
```suggestion
            (schema_name, table_name)
```

File: src/meta/src/manager/catalog/mod.rs [hunk 3]
```diff
@@ -414,9 +414,7 @@ where
         let core = &mut self.core.lock().await.database;
         let key = (table.database_id, table.schema_id, table.name.clone());
 
-        if core.has_table(table) {
-            Err(MetaError::catalog_duplicated("table", &table.name))
-        } else if core.has_in_progress_creation(&key) {
+        if core.has_in_progress_creation(&key) {
```
Comment (HuaHuaY) [inline src/meta/src/manager/catalog/mod.rs:417]: Forget to check duplicate names?

**Review by HuaHuaY — APPROVED** [review]
LGTM

**Review by yezizp2012 — APPROVED** [review]

**Pull request merged**